packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -146,6 +146,170 @@ class _OpenUpwardsPageTransition extends StatelessWidget {
   }
 }
 
+// Zooms and fades a new page in, zooming out the previous page. This transition
+// is designed to match the Android 10 activity transition.
+class _ZoomPageTransition extends StatefulWidget {
+  const _ZoomPageTransition({
+    Key key,
+    this.animation,
+    this.secondaryAnimation,
+    this.child,
+  }) : super(key: key);
+
+  // The scrim obscures the old page by becoming increasingly opaque.
+  static final Tween<double> _scrimOpacityTween = Tween<double>(
+    begin: 0.0,
+    end: 0.60,
+  );
+
+  // A curve sequence similar to the 'fastOutExtraSlowIn' curve used in the
+  // native transition.
+  static final List<TweenSequenceItem<double>> fastOutExtraSlowInTweenSequenceItems = <TweenSequenceItem<double>>[
+    TweenSequenceItem<double>(
+      tween: Tween<double>(begin: 0.0, end: 0.4)
+        .chain(CurveTween(curve: const Cubic(0.05, 0.0, 0.133333, 0.06))),
+      weight: 0.166666,
+    ),
+    TweenSequenceItem<double>(
+      tween: Tween<double>(begin: 0.4, end: 1.0)
+        .chain(CurveTween(curve: const Cubic(0.208333, 0.82, 0.25, 1.0))),
+      weight: 1.0 - 0.166666,
+    ),
+  ];
+  static final TweenSequence<double> _scaleCurveSequence = TweenSequence<double>(fastOutExtraSlowInTweenSequenceItems);
+  static final FlippedTweenSequence _flippedScaleCurveSequence = FlippedTweenSequence(fastOutExtraSlowInTweenSequenceItems);
+
+  final Animation<double> animation;
+  final Animation<double> secondaryAnimation;
+  final Widget child;
+
+  @override
+  __ZoomPageTransitionState createState() => __ZoomPageTransitionState();
+}
+
+class __ZoomPageTransitionState extends State<_ZoomPageTransition> {
+  AnimationStatus _currentAnimationStatus;
+  AnimationStatus _lastAnimationStatus;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.animation.addStatusListener((AnimationStatus animationStatus) {
+      _lastAnimationStatus = _currentAnimationStatus;
+      _currentAnimationStatus = animationStatus;
+    });
+  }
+
+  bool get _transitionWasInterrupted {
+    bool wasInProgress = false;
+    bool isInProgress = false;
+
+    switch (_currentAnimationStatus) {
+      case AnimationStatus.completed:
+      case AnimationStatus.dismissed:
+        isInProgress = false;
+        break;
+      case AnimationStatus.forward:
+      case AnimationStatus.reverse:
+        isInProgress = true;
+        break;
+    }
+    switch (_lastAnimationStatus) {
+      case AnimationStatus.completed:
+      case AnimationStatus.dismissed:
+        wasInProgress = false;
+        break;
+      case AnimationStatus.forward:
+      case AnimationStatus.reverse:
+        wasInProgress = true;
+        break;
+    }
+    return wasInProgress && isInProgress;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final Animation<double> _forwardScrimOpacityAnimation = _ZoomPageTransition._scrimOpacityTween
```

Comment:
You could write these animations in terms of [Animation.drive](https://api.flutter.dev/flutter/animation/Animation/drive.html). For example:
```dart
final Animation<double> _forwardScrimOpacityAnimation = widget.animation.drive(
  CurveTween(curve: const Interval(0.2075, 0.4175)),
);
```

Reply:
Is there a difference having these animations in terms of `Animation.drive` vs using `Tween.animate`?

Reply:
Animation.drive is consider to be a little easier to read. They're functionally the same.